### DISCUSSION TITLE
Web Workers work perfectly fine on all major browsers except Safari

### DISCUSSION THREAD

**Discussion by ObaidNadeem** [webpack/webpack#17287] [Q&A]
So I am using web worker in my WEBPACK 5 project, works fine on all major browsers, except, ofcourse Safari.

By checking on caniuse.com, Safari does has support for web workers since v5, but it keeps throwing this error in console:

```unhandled promise rejection, reference error, cant find variable worker safari error```

According to caniuse, Web Workers do have support in Safari.

This is what I am doing,

```
// in main js file

let worker = new Worker(new URL('./worker.js', import.meta.url));
worker.postMessage("start")
worker.onmessage = function (event) { do task }

// in worker.js

// Start on message receive
self.onmessage = function (event) {

            // do something here

            // Send the updated data back to the main thread
            self.postMessage({
              // send data
            });
}; 
```
What could be the issue ?

**Comment by alexander-akait** [comment]
Without reproducible example/more information no one can help, sorry

**Reply by ObaidNadeem** [reply]
Sure, what do you wanna know.

**Reply by alexander-akait** [reply]
Reproducible example and Safari versions

**Comment by LalitAgrawala004** [comment]
@ObaidNadeem did you find the fix for this?

**Reply by ObaidNadeem** [reply]
I believe I was using nested workers at that time and it wasn't supported by Safari. 

A worker.js should simply work.

**Reply by temeddix** [reply]
Hi, may I ask if Safari still doesn't support nested web workers?
